Documentation/op-guide/etcd3_alert.rules.yml

```diff
@@ -55,26 +55,36 @@ groups:
       description: on etcd instance {{ $labels.instance }} gRPC requests to {{ $labels.grpc_method
         }} are slow
       summary: slow gRPC requests
-  - record: instance:fd_utilization
-    expr: process_open_fds / process_max_fds
-  - alert: FdExhaustionClose
-    expr: predict_linear(instance:fd_utilization[1h], 3600 * 4) > 1
+  - alert: HighNumberOfFailedHTTPRequests
+    expr: sum(rate(etcd_http_failed_total{job="etcd"}[5m])) BY (method) / sum(rate(etcd_http_received_total{job="etcd"}[5m]))
```

Comment:
why do we care about http received total if this is for etcd3 alerts?

Reply:
I've just synced it from prometheus-operator. This repo have original jsonnet file: https://github.com/etcd-io/etcd/blob/master/Documentation/etcd-mixin/mixin.libsonnet added in https://github.com/etcd-io/etcd/pull/9640, I'll fix that PR according to it and will update prometheus-operator then.

Reply:
xiang90, it's it's for etcd3 http requests, not about k8s-wise http.

Reply:
> xiang90, it's it's for etcd3 http requests, not about k8s-wise http.

etcd3's API is gRPC API not http API

Reply:
They are written there: https://github.com/etcd-io/etcd/blob/master/etcdserver/api/v2http/metrics.go
To me it looks like this is still in use, otherwise it won't be there. I'll be waiting for your decision to keep or delete alerts on these metrics.
On new prometheus-operator helm chart I've imported these and immediately found someone with the problem, e.g. metrics are written in his case: https://github.com/helm/charts/pull/9103#issuecomment-437927250

Reply:
As per https://github.com/helm/charts/pull/9103#issuecomment-438304831 metrics are present, but could false-positive on 404 because of the way kubelet works. In my opinion we should leave these alerts, but excluded 404 errors from count, I'll do that in next commit.

Reply:
Metrics are present in real-life installations, I've excluded 404 because of false-positive. Resolving the conversation.